# DP: Description: Fix build when __thread is not available.
# DP: Dpatch author: Michael Banck <mbanck@debian.org>
# DP: Patch author: Thomas Schwinge
# DP: Upstream status: Rejected
# DP: Date: 2005-08-28

2005-08-28  Thomas Schwinge  <schwinge@nic-nac-project.de>

	malloc/memusage.c: Only use __thread if USE___THREAD.


Index: malloc/memusage.c
===================================================================
RCS file: /cvs/glibc/libc/malloc/memusage.c,v
retrieving revision 1.12
diff -u -r1.12 memusage.c
--- malloc/memusage.c	20 Aug 2005 01:12:37 -0000	1.12
+++ malloc/memusage.c	27 Aug 2005 23:43:02 -0000
@@ -83,7 +83,11 @@
 static memusage_cntr_t decreasing_mremap;
 static memusage_size_t current_heap;
 static memusage_size_t peak_use[3];
+#if USE___THREAD
 static __thread uintptr_t start_sp;
+#else
+static uintptr_t start_sp;
+#endif
 
 /* A few macros to make the source more readable.  */
 #define peak_heap	peak_use[0]
